lib/usecases/run/GetAllRunsUseCase.js

```diff
@@ -207,8 +207,24 @@ class GetAllRunsUseCase {
                 const lhcPeriodsList = lhcPeriods
                     .split(SEARCH_ITEMS_SEPARATOR)
                     .map((lhcPeriod) => lhcPeriod.trim());
-                filteringQueryBuilder.where('lhcPeriod').oneOf(...lhcPeriodsList);
+
+                const runsWithExpectedLHCPeriod = await RunRepository.findAll({
+                    attributes: ['id'],
+                    include: [
+                        {
+                            association: 'lhcPeriod',
+                            required: true,
+                            attributes: [],
+                            where: {
+                                name: { [Op.in]: lhcPeriodsList },
+                            },
+                        },
+                    ],
+                });
+
+                filteringQueryBuilder.where('id').oneOf(...runsWithExpectedLHCPeriod.map(({ id }) => id));
```

Comment:
You can probably apply directly the include on the filterQueryBuilder, as for the "or" case of tags.

Reply:
right, modified